### PR TITLE
Adding tests and implementation for isBool

### PR DESCRIPTION
**Pull request by ipetkov**
My iOS application interfaces with a JSON api which sometimes returns `true` or `false` values, which are placed inside an `NSNumber` wrapper when the JSON is parsed.

Objective C BOOLs are actually defined as `signed char` so calling `[num boolValue]` on an `NSNumber` with any other primitive besides `BOOL` will not yield the expected `YES` or `NO` conversions. Constantly using `isKindOfClass` checks is not an aesthetically pleasing alternative.

I've added the method `isBool` to double check if an `NSNumber` was created via `[NSNumber numberWithBool:bool]` and not some other primitive type.

### CODE REVIEW

File: Underscore/Underscore.m [hunk 1]
```diff
@@ -57,6 +57,13 @@ + (UnderscoreTestBlock)isArray
     };
 }
 
++ (UnderscoreTestBlock)isBool
+{
+    return ^BOOL (id obj) {
+        return [obj isKindOfClass:[[NSNumber numberWithBool:YES] class]];
```
Comment (robb) [inline Underscore/Underscore.m]: Relying on `__NSCFBoolean` or what the class is called seems to be a bit fragile.

What about something like

``` objc
[obj isKindOfClass:NSNumber.class] && strcmp([obj objCType], @encode(BOOL)) == 0
```

?

**Comment by robb** [comment]
Makes a lot of sense :+1: Just one comment

File: Underscore/Underscore.m (hunk 1 above)
Comment (toto) [inline Underscore/Underscore.m]: Doen't this also pose a problem if people want to run this with `[[NSNumber numberWithInt:1] boolValue]`?

Comment (ipetkov) [inline Underscore/Underscore.m]: @robb Ah you're right, that's definitely a stronger check!

@toto I, personally, was looking for a stricter type check to distinguish `[NSNumber numberWithBool:YES]` vs `[NSNumber numberWithInt:1]`.

Calling `boolValue` on an `NSNumber` with any integer besides 0 will produce `YES` meaning `[[NSNumber numberWithInt:1] boolValue]` is the equivalent of `[[NSNumber numberWithInt:1] intValue] != 0`.

In my opinion, the two `NSNumber` types should be treated as separate, and anyone wishing to use a more agnostic approach can just use `_.isNum(num) && [num boolValue]` as a filter.

**Pull request merged**